### PR TITLE
chore: upgrade dependencies

### PR DESCRIPTION
**Pull request by Josh-Cena**
## Pre-flight checklist

- [ ] I have read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#pull-requests).
- [ ] **If this is a code change**: I have written unit tests and/or added dogfooding pages to fully verify the new behavior.
- [ ] **If this is a new API or substantial change**: the PR has an accompanying issue (closes #0000) and the maintainers have approved on my working plan.

## Motivation

Due to the shear number of dependencies needing update, this PR only does minor/patch updates. All major updates go into a separate PR to avoid breaks as much as possible.

### Test links

Deploy preview: https://deploy-preview-_____--docusaurus-2.netlify.app/

### CODE REVIEW

File: website/testCSSOrder.mjs [hunk 1]
```diff
@@ -31,6 +31,7 @@ const EXPECTED_CSS_MARKERS = [
   // See https://github.com/facebook/docusaurus/pull/6222
   '.markdown>h2',
   '.button--outline.button--active',
+  '.DocSearch-Hit-content-wrapper',
```
Comment (Josh-Cena) [inline website/testCSSOrder.mjs:34]: For some mysterious reason, this has reverted to pre-#8064 behavior

**Comment by Josh-Cena** [comment]
cc @shortcuts We are getting these type checking failures:

```
Error: ../node_modules/@algolia/autocomplete-preset-algolia/dist/esm/types/algoliasearch.d.ts(7,74): error TS2339: Property 'liteClient' does not exist on type 'typeof import("/home/runner/work/docusaurus/docusaurus/node_modules/algoliasearch/lite")'.
Error: ../node_modules/@algolia/autocomplete-preset-algolia/dist/esm/types/algoliasearch.d.ts([8](https://github.com/facebook/docusaurus/actions/runs/3878383730/jobs/6614488154#step:6:9),75): error TS2339: Property 'searchClient' does not exist on type 'typeof import("/home/runner/work/docusaurus/docusaurus/node_modules/@algolia/client-search/dist/client-search")'.
Error: ../node_modules/@algolia/autocomplete-preset-algolia/dist/esm/types/algoliasearch.d.ts(20,23): error TS26[9](https://github.com/facebook/docusaurus/actions/runs/3878383730/jobs/6614488154#step:6:10)4: Namespace '"/home/runner/work/docusaurus/docusaurus/node_modules/algoliasearch/lite"' has no exported member 'LegacySearchMethodProps'.
Error: ../node_modules/@algolia/autocomplete-preset-algolia/dist/esm/types/algoliasearch.d.ts(24,23): error TS2694: Namespace '"/home/runner/work/docusaurus/docusaurus/node_modules/algoliasearch/lite"' has no exported member 'SearchForFacetValuesResponse'.
Error: ../node_modules/@algolia/autocomplete-preset-algolia/dist/esm/types/algoliasearch.d.ts(28,23): error TS2694: Namespace '"/home/runner/work/docusaurus/docusaurus/node_modules/algoliasearch/lite"' has no exported member 'SearchResponse'.
Error: ../node_modules/@algolia/autocomplete-preset-algolia/dist/esm/types/algoliasearch.d.ts(32,23): error TS2694: Namespace '"/home/runner/work/docusaurus/docusaurus/node_modules/algoliasearch/lite"' has no exported member 'HighlightResult'.
Error: ../node_modules/@algolia/autocomplete-preset-algolia/dist/esm/types/algoliasearch.d.ts(36,23): error TS2694: Namespace '"/home/runner/work/docusaurus/docusaurus/node_modules/algoliasearch/lite"' has no exported member 'SnippetResult'.
```

Does that look like a bug in the Algolia code base?

**Comment by shortcuts** [comment]
oops it seems to be an upstream update that wasn't detected on the DocSearch side by renovate 🤔 releasing the update in 3.3.2

**Comment by shortcuts** [comment]
update: fixed in 3.3.2

**Comment by Josh-Cena** [comment]
Thank you—that was really helpful!

**Pull request merged**